app/controllers/concerns/enforce_eligibility_question_order.rb

```diff
@@ -18,6 +22,11 @@ def current_page_is_allowed?
     current_position && order && current_position <= order
   end
 
+  def ineligible_application_without_skip_additional_questions
+    eligibility_check&.status == :eligibility &&
```

Comment:
```suggestion
    eligibility_check&.status == :eligibility
```

The status of `:eligibility` includes both eligible and ineligible checks - if the check is eligible I don't think we want to redirect to the ineligible page.